public/offline.html

```diff
@@ -250,14 +291,30 @@ <h4>(pick a color below && start drawing!)</h4>
       firstX = null
       firstY = null
     }
-
     canvas.addEventListener('mouseup', exit)
     canvas.addEventListener('mouseleave', exit)
     canvas.addEventListener('touchend', exit)
 
+    function endPaint(event) {
+      if (paint) {
+        addClick(event, this)
+        draw()
+      }
+    }
     canvas.addEventListener('mousemove', endPaint)
     canvas.addEventListener('touchmove', endPaint)
 
+    const paletteSelector = document.querySelector(".paletteSelector")
+    paletteSelector.onchange = (event) => {
+      activePalette = event.target.value
+      render(palettes[activePalette])
+    }
+
+    // Initialization when page is ready to load, runs once
+    (function() {
```

Comment:
Introduces convention of putting all page initialization code (aside from binding event handlers) into one orderly location.